src/sinks/util/builder.rs

```diff
@@ -0,0 +1,138 @@
+use std::{fmt, future::Future, io, num::NonZeroUsize, pin::Pin, sync::Arc, time::Duration};
+
+use futures_util::Stream;
+use tower::Service;
+use vector_core::{
+    buffers::{Ackable, Acker},
+    event::{Event, EventStatus, Finalizable},
+    partition::Partitioner,
+    stream::{
+        batcher::{Batcher, ExpirationQueue},
+        driver::Driver,
+    },
+};
+
+use super::{encoding::Encoder, Compression, Compressor, ConcurrentMap, RequestBuilder};
+
+impl<T: ?Sized> SinkBuilderExt for T where T: Stream {}
+
+pub trait SinkBuilderExt: Stream {
```

Comment:
Seems like this could be a part of `vector_core::stream`.

Reply:
Going to punt on moving any of this code until after the PR is merged, as it's grown a bit long in the tooth and I want to merge it sooner rather than later.

Reply:
Would you mind filing a follow-up issue?